docs(gcp): revamp installation (#28)
openclarity/docs.openclarity.io: content/docs/vmclarity/getting-started/gcp/_index.md

---
title: Deploy on GCP
weight: 120
---

> These instructions are for installing VMClarity on [GCP (Google Cloud Platform)](https://cloud.google.com).

Reviewer: ```suggestion
To install VMClarity on [Google Cloud Platform (GCP)](https://cloud.google.com), complete the following steps.
```

## Prerequisites

> VMClarity on GCP can be installed currently via the CLI, so you have to have
> [gcloud](https://cloud.google.com/sdk/gcloud) on your computer available beforehand. For further instructions about
> installing and configuring gcloud check the [official installation guide](https://cloud.google.com/sdk/docs/install).

Reviewer: ```suggestion
You can install VMClarity using the CLI, so you have to have [gcloud](https://cloud.google.com/sdk/gcloud) on your computer available beforehand. For details on installing and configuring gcloud, see the [official installation guide](https://cloud.google.com/sdk/docs/install).
```

   | `externalDBHost`                |          |                                                                             | Hostname or IP for the external DB. Only required if the DatabaseToUse is External. |
   | `externalDBPort`                |          |                                                                             | Port for the external DB. Only required if the DatabaseToUse is External.           |

1. Deploy VMClarity using gcloud deployment-manager.

   ```shell
   gcloud deployment-manager deployments create <vmclarity deployment name> --config vmclarity-config.yaml
   ```

## Access VMClarity UI

1. Open an SSH tunnel to the VMClarity server with gcloud. For further information on how to create an SSH connection
   with gcloud to one of your instances check the [official page](https://cloud.google.com/compute/docs/connect/standard-ssh#gcloud).

   ```shell
   gcloud compute ssh --project=<project id> --zone=<zone name> <name of your VM> -- -NL 8080:localhost:80
   ```

1. Open the VMClarity UI in your browser at [http://localhost:8080](http://localhost:8080).

>     <p align="center" width="100%">
>         <img width="75%" src="/img/vmclarity-ui-1.png">
>     </p>

Reviewer: ```suggestion
![VMClarity UI Dashboard](/img/vmclarity-ui-1.png)
```
